src/activate.ts

```
@@ -440,7 +440,16 @@ async function swapActivationRuntimeRoot(
   } catch (error) {
     if (hadRuntimeRoot && !(await pathExists(runtimeRoot))) {
-      await rename(backupRuntimeRoot, runtimeRoot).catch(() => undefined);
+      try {
+        await rename(backupRuntimeRoot, runtimeRoot);
+      } catch (rollbackError) {
+        throw new AggregateError(
+          [error, rollbackError],
+          "Activation failed and the runtime root rollback also failed — " +
+            "the runtime root may be missing. Restore it manually from the " +
+            `backup at '${backupRuntimeRoot}' if present.`,
+        );
```

ESLint

[error] There is no `cause` attached to the symptom error being thrown.

(preserve-caught-error)